python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -483,9 +520,30 @@ def debug(self):
             logger.error(result.get('Message', ''))
 
     @staticmethod
-    def _getStateTime(job, status):
+    def _bulk_updateStateTime( jobStateDict ):
+        """ This performs the same as the _getStateTime method but loops over a list of job ids within the DIRAC namespace (much faster)
```

Comment:
Sorry, I assumed the PR was ready for review as you were asking to merge it. I will wait until you are ready next time. Let me know when you want this one reviewing.